src/Content/Backend/Solution/Monaco.Template.Backend.Api/Controllers/CompaniesController.cs

```diff
@@ -19,35 +20,28 @@ namespace Monaco.Template.Backend.Api.Controllers;
 
 [Route("api/v{apiVersion:apiVersion}/[controller]")]
 [ApiController]
-public class CompaniesController : ControllerBase
```

Comment:
How do we feel about primary constructor parameters not being `readonly`? Resulting in the injected dependencies becoming mutable. In reality this is probably a non issue but the former syntax did prevent us from overwriting the dependency accidently due to the field being `readonly`.

Further - I personally prefer the separation of class scoped fields named for example as `_mediator` being clearly different from method scoped parameters named for example as `mediator`. So the new primary constructor syntax makes this less distinct.

Reply:
It's a valid discussion, indeed.
I'm not on the computer right now, but will take a look later at some point and will try to think about what approach would be best.

TBH, most of the primary constructor refactors are the result of some R# automation, so better to give it a more proper thought.

Reply:
To add to the discussion, the above points have since been raised in this video: https://www.youtube.com/watch?v=IABO8O9cZOw

Reply:
Oh, yeah, just yesterday I've seen it and I was due to comment about it here as well.

I've been thinking about this and I had a couple of the options Nick already showed over there going around in my mind.

For example, one alternative I was thinking about was to initialize our own fields like:

```csharp
public class Class1(int fieldA)
{
    private readonly int _fieldA = fieldA;
}
```

This would allow us to still keep a private readonly field, but with the primary constructors things changed in the way that the primary constructor variables are actually parameters, they're not fields and they're not public members or anything like that. If you hold your mouse over them the IDE will clearly identify them as parameters. Problem is, they can still be used in most places around a class (except in derived classes) and could be confused and used instead of the actual fields in the class where they're declared. This could potentially lead to more confusions and result in misbehaviors.

On the other side, using only the parameters will be very constrained if we need to have protected fields to be used in derived classes, so it will instead be required to explicitly implement those protected fields in a similar way as what I mentioned above.

A last alternative is to not do any changes and keep using the fields as we were doing until now and don't implement primary constructors at all like Nick finally suggested.

I liked the idea of reducing our classes boilerplate and streamline the code, but seems like what Nick was saying: it still looks like a half-baked feature that misses few useful things to go all in with it...

What are your thoughts, @matthewtoghill?

Reply:
I tend to agree with Nick that the feature is half-baked right now. I hope it will be improved more in the future to allow the primary constructor (PC) parameters/properties to be decorated with a `readonly` modifier so that the refactor to use PCs provides an equal outcome to the non-PC syntax.

I agree as well that it may be confusing to just assign the parameters to new private readonly fields as the PC parameters will still be available for use within the class and their use may be mix and matched in error. 

For now I think I'll be avoiding it for dependency injection.

Reply:
Yeah, I think I'd better rollback those changes back to leave it using the original implementation with private/protected readonly fields.

Reply:
Done.